plugin/trino-elasticsearch/src/main/java/io/trino/plugin/elasticsearch/decoders/RawJsonDecoder.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.elasticsearch.decoders;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.airlift.json.ObjectMapperProvider;
+import io.airlift.slice.Slices;
+import io.trino.spi.TrinoException;
+import io.trino.spi.block.BlockBuilder;
+import io.trino.spi.type.VarcharType;
+import org.elasticsearch.search.SearchHit;
+
+import java.util.function.Supplier;
+
+import static io.trino.spi.StandardErrorCode.TYPE_MISMATCH;
+import static io.trino.spi.type.VarcharType.VARCHAR;
+import static java.lang.String.format;
+import static java.util.Objects.requireNonNull;
+
+public class RawJsonDecoder
+        implements Decoder
+{
+    public static final int MAX_LENGTH = Integer.MAX_VALUE - 1;
+    public static final VarcharType RAW_JSON = VarcharType.createVarcharType(MAX_LENGTH);
```

Comment:
This is a bit of a hack. I am making a varchar type of a different size which basically functions as another type to specify when we are dealing with raw json field vs a regular string, long, other field.

Does it make more sense to make an actual raw json type?

Reply:
based on https://github.com/trinodb/trino/issues/7308#issuecomment-800670713, I think using `VARCHAR` type makes sense. but I think it'd be cleaner to propagate the `asRawJson` flag through column handles and utilize it while creating the decoder (instead of trying to encode this information through varchartype length). thoughts?

Reply:
I think that makes sense. @aalbu what do you think? This keeps us away from adding a new type.

Reply:
Yes, that's a good suggestion.  Simpler than having a specific type.

Reply:
@aalbu, I updated based on @phd3's suggestion. We now maintain the property tree on the es column handle. 

Another suggestions from Alex, is to not allow `isArray` flag with the `asRawJson` flag as the syntax between Trino arrays and json arrays can get ugly and confusing.

CC: @martint